numba/cuda/tests/cudapy/test_atomics.py

```diff
@@ -530,6 +569,111 @@ def kernel(x):
+
+    def test_atomic_nanmax_float32(self):
+        self.check_atomic_nanmax(dtype=np.float32, lo=-65535, hi=65535)
+
+    def test_atomic_nanmax_double(self):
+        self.check_atomic_nanmax(dtype=np.float64, lo=-65535, hi=65535)
+
+    def test_atomic_nanmax_double_shared(self):
+        vals = np.random.randint(0, 32, size=32).astype(np.float64)
+        vals[1::2] = np.nan
+        res = np.array([0], dtype=vals.dtype)
+        cuda_func = cuda.jit('void(float64[:], float64[:])')(atomic_nanmax_double_shared)
+        cuda_func[1, 32](res, vals)
+
+        gold = np.nanmax(vals)
+        np.testing.assert_equal(res, gold)
+
+    def test_atomic_nanmax_double_oneindex(self):
+        vals = np.random.randint(0, 128, size=32).astype(np.float64)
+        vals[1::2] = np.nan
+        res = np.zeros(1, np.float64)
+        cuda_func = cuda.jit('void(float64[:], float64[:])')(
+            atomic_max_double_oneindex)
+        cuda_func[1, 32](res, vals)
+
+        gold = np.nanmax(vals)
+        np.testing.assert_equal(res, gold)
+
+    # nanmin tests
+    def check_atomic_nanmin(self, dtype, lo, hi):
+        vals = np.random.randint(lo, hi, size=(32, 32)).astype(dtype)
+        vals[1::2] = np.nan
+        res = np.array([65535], dtype=vals.dtype)
+        cuda_func = cuda.jit(atomic_nanmin)
+        cuda_func[32, 32](res, vals)
+
+        gold = np.nanmin(vals)
+        np.testing.assert_equal(res, gold)
+
+    def test_atomic_nanmin_int32(self):
+        self.check_atomic_nanmin(dtype=np.int32, lo=-65535, hi=65535)
+
+    def test_atomic_nanmin_uint32(self):
+        self.check_atomic_nanmin(dtype=np.uint32, lo=0, hi=65535)
+
+    @skip_unless_cc_32
+    def test_atomic_nanmin_int64(self):
+        self.check_atomic_nanmin(dtype=np.int64, lo=-65535, hi=65535)
+
+    @skip_unless_cc_32
+    def test_atomic_nanmin_uint64(self):
+        self.check_atomic_nanmin(dtype=np.uint64, lo=0, hi=65535)
+
+    def test_atomic_nanmin_float(self):
+        self.check_atomic_nanmin(dtype=np.float32, lo=-65535, hi=65535)
+
+    def test_atomic_nanmin_double(self):
+        self.check_atomic_nanmin(dtype=np.float64, lo=-65535, hi=65535)
+
+    def test_atomic_nanmin_double_shared(self):
+        vals = np.random.randint(0, 32, size=32).astype(np.float64)
+        vals[1::2] = np.nan
+        res = np.array([32], dtype=vals.dtype)
+        cuda_func = cuda.jit('void(float64[:], float64[:])')(atomic_nanmin_double_shared)
+        cuda_func[1, 32](res, vals)
+
+        gold = np.nanmin(vals)
+        np.testing.assert_equal(res, gold)
+
+    def test_atomic_nanmin_double_oneindex(self):
+        vals = np.random.randint(0, 128, size=32).astype(np.float64)
+        vals[1::2] = np.nan
+        res = np.array([128], np.float64)
+        cuda_func = cuda.jit('void(float64[:], float64[:])')(
+            atomic_min_double_oneindex)
+        cuda_func[1, 32](res, vals)
+
+        gold = np.nanmin(vals)
+        np.testing.assert_equal(res, gold)
+
```

Comment:
Wonder if tests similar to https://github.com/numba/numba/pull/5941/files#diff-265d0d9cde3f18a334cc9633801aa6f1R509-R568 should be added?

Reply:
I think we should test the return value just in case the code drift.